src/components/contextual/pages/pools/UnstakedPoolsTable.vue

```diff
@@ -155,16 +153,12 @@ function handleModalClose() {
         @trigger-stake="handleStake"
       />
     </BalStack>
-    <StakingProvider
+    <StakePreviewModal
       v-if="stakePool"
-      :poolAddress="getAddressFromPoolId(stakePool.id)"
```

Comment:
If I'm not wrong, `stakePool.id` will be undefined until the `stakePool` is created after calling   `setPoolAddress(pool.address)`:

```ts
 handleStake(pool: Pool) {
  setPoolAddress(pool.address);
  showStakeModal.value = true;
  stakePool.value = pool;
} 
```

That means that we can avoid initialising poolAddress when calling `StakingProvider`. Am I missing something?